pkg/build/controller/factory/factory.go

```diff
@@ -96,9 +96,13 @@ func (factory *BuildControllerFactory) Create() controller.RunnableController {
 			build := obj.(*buildapi.Build)
 			err := buildController.HandleBuild(build)
 			if err != nil {
-				build.Status.Message = err.Error()
-				if err := buildController.BuildUpdater.Update(build.Namespace, build); err != nil {
-					glog.V(2).Infof("Failed to update status message of Build %s/%s: %v", build.Namespace, build.Name, err)
+				msg := err.Error()
+				// Update the build status message only if it changed.
+				if build.Status.Message != msg {
+					build.Status.Message = msg
+					if err := buildController.BuildUpdater.Update(build.Namespace, build); err != nil {
+						glog.V(2).Infof("Failed to update status message of Build %s/%s: %v", build.Namespace, build.Name, err)
+					}
```

Comment:
Can't talk about the future, but for now that's the only place that will update the message in the context of an error. In https://github.com/openshift/origin/pull/4872, pretty much by design, I tried to find a single place to report the error instead of handling it in multiple scattered places, which would be more (human) error prone IHMO.

I also looked at adding server logic to prevent doing an update that doesn't change anything, but looking around it didn't seem like the way to go.